Mathlib/Algebra/Category/GroupCat/Adjunctions.lean

```diff
@@ -60,7 +60,7 @@ theorem free_obj_coe {α : Type u} : (free.obj α : Type u) = FreeAbelianGroup 
   rfl
 #align AddCommGroup.free_obj_coe AddCommGroupCat.free_obj_coe
 
-@[simp]
+@[simp, nolint simpNF]
```

Comment:
This one doesn't look like the `simps` bug; what goes wrong with this lemma?

Reply:
```lean
 Left-hand side simplifies from
  ↑(AddCommGroupCat.free.map f) x
to
  ↑(AddCommGroupCat.free.map f) x
using
  simp only [@AddCommGroupCat.free_obj_coe]
```
where the first is obfuscated 
```lean
@FunLike.coe (↑(AddCommGroupCat.free.obj α) →+ ↑(AddCommGroupCat.free.obj β)) (↑(AddCommGroupCat.free.obj α))
  (fun a ↦ (fun x ↦ ↑(AddCommGroupCat.free.obj β)) a) AddHomClass.toFunLike (AddCommGroupCat.free.map f) x :
(fun x ↦ ↑(AddCommGroupCat.free.obj β)) x
```
and the latter is the cleaner
```lean
@FunLike.coe (FreeAbelianGroup α →+ FreeAbelianGroup β) (FreeAbelianGroup α) (fun a ↦ FreeAbelianGroup β)
  AddHomClass.toFunLike (AddCommGroupCat.free.map f) x :
FreeAbelianGroup β
```

Reply:
Left a comment.